src/rng.rs

```diff
@@ -1,32 +1,61 @@
 use crate::Vec;
-use algebra::{FromBytes, ToBytes, Field};
+use algebra::{FromBytes, ToBytes, to_bytes, Field};
 use std::{
     convert::TryInto,
-    marker::PhantomData
+    marker::PhantomData,
 };
 use digest::{generic_array::GenericArray, Digest};
 use rand_chacha::ChaChaRng;
 use rand_core::{RngCore, SeedableRng};
 use rand::Rng;
+use crate::error::Error;
 
-/// General trait for `SeedableRng` that refreshes its seed by hashing together the previous seed
-/// and the new seed material.
-// TODO: later: re-evaluate decision about ChaChaRng
-pub trait FiatShamirRng: RngCore {
+/// General trait for an updatable seed to be used to initialize a `FiatShamirRng`.
+pub trait FiatShamirRngSeed {
+    /// Output type of the seed, to be consistent with the seed type accepted
+    /// by the `from_seed` function of FiatShamirRng.
+    type FinalizedSeed: Sized + Clone;
+
+    /// Error type
+    type Error: std::error::Error + From<Error>;
+
+    /// Maximum number of elements allowed to be part of this seed
```

Comment:
Lets add to the comment "(limited by our constraint to have just a single length and type encoding field element)"

Reply:
Let us just remove it, both constants are part of the add_bytes implementation.